tldw_Server_API/app/core/RAG/rag_service/analytics_db.py

```
@@ -491,8 +566,20 @@ def _fetchall(
+        backend = self.backend
+        previous_backend = getattr(self._local, "backend_pin", None)
+        self._local.backend_pin = backend
+        try:
+            with backend.transaction() as conn:
+                yield conn
+        finally:
+            if previous_backend is None:
+                try:
+                    del self._local.backend_pin
+                except AttributeError:
+                    pass
```

Ruff (0.15.10)

[warning] Use `contextlib.suppress(AttributeError)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(AttributeError): ...`

(SIM105)